nodejs/week2/Mealsharing/src/backend/routes/meal.js

```diff
@@ -0,0 +1,80 @@
+const meals = require("../data/meals.json");
+const express = require("express");
+const router = express.Router();
+router.get("/", (req, res) => {
+          // default route to access the all meals
+          if (!meals) {
+                    res.status(404).send("Data does not exists");
+          }
+          res.send(meals);
+});
+router.get("/meals/:id", (req, res) => {
+          // route with id
+          const getId = parseInt(req.params.id);
+          const getMeal = meals.filter((meal) => meal.id === getId);
+          if (!getMeal.length) {
+                    res.status(404).send("Meal does not  exists of this id");
+          }
+          res.send(getMeal);
+});
+
+router.get("/meals", (req, res) => {
+          // Query method
+          const keyValue = Object.keys(req.query);
+          if (keyValue[0] == "maxPrice") {
+                    //using maxPrice
+                    const getPrice = parseInt(req.query.maxPrice);
+                    const getMeal = meals.filter(
+                              (meal) => meal.price < getPrice
+                    );
+                    if (!getMeal.length) {
+                              res.status(404).send(
+                                        "Meal does not  exists of this price"
+                              );
+                    }
+                    res.send(getMeal);
+          } else if (keyValue[0] == "title") {
+                    //using title
+                    const getTitle = req.query.title;
+                    const exp = getTitle.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+                    const pattern = new RegExp(
+                              "(?:^|\\W)" + exp + "(?:$|\\W)",
+                              "i"
+                    );
+
+                    const getMeal = meals.filter((meal) =>
+                              pattern.test(meal.title)
+                    );
```

Comment:
Good that you have found a way to find a title in meal.title,
but there is also another way to solve this problem.
You could write like
```
                   const getMeal = meals.filter(item  => {
                        return item.title.toLowerCase().includes(getTitle.toLowerCase())
                    })
```